libs/portals/admin/ids-admin/src/utils/checkEnvironmentsSync.ts

```diff
@@ -0,0 +1,28 @@
+import { DynamicEnvironmentResult } from '../types/dynamicEnvironmentResult'
+
+/**
+ * Checks if the given variables are in sync across the given environments.
+ * @param environments
+ * @param variables - The developer is responsible to provide the correct subset of variables.
+ */
+export const checkEnvironmentsSync = <T extends DynamicEnvironmentResult<T>>(
+  environments: T[],
+  variables: Array<keyof T>,
+) => {
+  const referenceEnvironment = environments[0]
+
+  for (const variableName of variables) {
+    const referenceValue = referenceEnvironment[variableName]
+
+    for (let i = 1; i < environments.length; i++) {
+      const currentEnvironment = environments[i]
+      const currentValue = currentEnvironment[variableName]
+
+      if (JSON.stringify(currentValue) !== JSON.stringify(referenceValue)) {
```

Comment:
Should be fine in this project, but would use lodash deepEqual in future projects to avoid errors with different orderings of properties:

```
JSON.stringify({a: 1, b: 2}) !== JSON.stringify({b: 2, a: 1})
```